src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomCubicWorldType.java

```diff
@@ -94,6 +94,12 @@ public ICubeGenerator createCubeGenerator(World world) {
     public boolean isCustomizable() {
         return true;
     }
+    
+    @Override
+    public void onGUICreateWorldPress() { 
+        GuiCreateWorld gui = (GuiCreateWorld) Minecraft.getMinecraft().currentScreen;
+        CustomGeneratorSettings.fromJson(CustomCubicGui.settingsJsonString).save(new File(Minecraft.getMinecraft().mcDataDir, "saves/"+gui.saveDirName+"/"));
```

Comment:
I don't really like the way you are accessing the world save location here. This could be different for many different reasons (it's not *that* hard to imaging a mod that lets you load worlds from any folder). There should be a way to do it after the World object already exists and get it normally. And that is kind of needed anyway for it to work on dedicated server.